### DISCUSSION TITLE
error about wgpu-hal

### DISCUSSION THREAD

**Discussion by aisuta** [hecrj/iced#1391] [Q&A] [answered]
a error occured when i compiling wgpu-hal

error: Metal API enabled on non-Apple OS. If your project is not using resolver="2" in Cargo.toml, it should.
  --> D:\Rust\cargo\registry\src\github.com-1ecc6299db9ec823\wgpu-hal-0.12.5\src\lib.rs:51:1
   |
51 | compile_error!("Metal API enabled on non-Apple OS. If your project is not using resolver=\"2\" in Cargo.toml, it should.");
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

cargo.toml 
[package]
name = "proj01"
version = "0.1.0"
edition = "2021"

[dependencies]
iced = "0.4"

code was running on windows10 
thank you

**Comment by tarkah** [comment, marked as answer]
Try putting `resolver = "2"` under `[package]` in your `Cargo.toml`

**Reply by aisuta** [reply]
thank you，but
when i use workspace to organize my code and create a new rust project in workspace， 
workspace root Cargo.toml
[workspace]
members=['proj-01']

Cargo.toml for proj-01
[package]
name = "proj01"
version = "0.1.0"
edition = "2021"
resolver = "2"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
iced = "0.4"

when I build my project , the same error occurred

however 
it succeed if there were no workspace configuration but a single proj-01

**Reply by tarkah** [reply]
When you have a workspace, you need to put the `resolver` key under `workspace` in your root `Cargo.toml`
